src/ncp/ncp_base.cpp

```diff
@@ -414,8 +416,8 @@ void NcpBase::HandleSendDone()
     if (mSendQueue.GetHead() != NULL)
     {
         Message &message(*mSendQueue.GetHead());
-        HandleDatagramFromStack(message);
         mSendQueue.Dequeue(message);
+        HandleDatagramFromStack(message);
```

Comment:
No. Dequeue  does not free the buffer.

I think that it handles the packets in the queue from the stack, and transmits to the host. The packets from the stack would enqueue, if the serial port is sending. Is my understanding right?